+sw_tests/+unit_tests/unittest_super.m

```diff
@@ -68,14 +68,20 @@ function verify_spinw_matrix(testCase, expected_matrix, actual_matrix, varargin)
         end
         function verify_spinwave(testCase, expected_spinwave, ...
                                  actual_spinwave, varargin)
-            % List of fields to test separately
-            rmfields = {'datestart', 'dateend', 'obj'};
+            % List of fields to test separately, only remove fields that
+            % exist
+            rmfields = intersect(fields(expected_spinwave), ...
+                                 {'datestart', 'dateend', 'obj'});
             testCase.verify_obj(rmfield(expected_spinwave, rmfields), ...
                                 rmfield(actual_spinwave, rmfields), varargin{:})
             for field = ["datestart", "dateend"]
-                testCase.assertTrue(isa(actual_spinwave.(field), 'char'));
+                if isfield(expected_spinwave, field)
+                    testCase.assertTrue(isa(actual_spinwave.(field), 'char'));
+                end
+            end
+            if isfield(expected_spinwave, 'obj')
```

Comment:
Presumably this is because `obj` is not aleways returned (only if `fitmode`==false) - would it be possible to add a comment here to clarify?